### PR TITLE
SLI-898 Add support for IaC domains in IntelliJ

### PR DESCRIPTION
**Pull request by nquinquenel**

### CODE REVIEW

File: src/main/resources/META-INF/plugin.xml [hunk 1]
```diff
@@ -195,6 +195,10 @@
         <externalAnnotator language="XML" implementationClass="org.sonarlint.intellij.editor.SonarExternalAnnotator"/>
         <externalAnnotator language="Scala" implementationClass="org.sonarlint.intellij.editor.SonarExternalAnnotator"/>
         <externalAnnotator language="Swift" implementationClass="org.sonarlint.intellij.editor.SonarExternalAnnotator"/>
+        <externalAnnotator language="cloudformation" implementationClass="org.sonarlint.intellij.editor.SonarExternalAnnotator"/>
+        <externalAnnotator language="docker" implementationClass="org.sonarlint.intellij.editor.SonarExternalAnnotator"/>
+        <externalAnnotator language="kubernetes" implementationClass="org.sonarlint.intellij.editor.SonarExternalAnnotator"/>
+        <externalAnnotator language="terraform" implementationClass="org.sonarlint.intellij.editor.SonarExternalAnnotator"/>
```
Comment (damien-urruty-sonarsource) [inline src/main/resources/META-INF/plugin.xml:201]: Are those 4 languages known by IntelliJ ? I was not able to find any doc

File: src/main/java/org/sonarlint/intellij/core/EmbeddedPlugins.kt [hunk 2]
```diff
@@ -58,7 +62,11 @@ object EmbeddedPlugins {
         EmbeddedPlugin(Language.RUBY, "Ruby", "sonar-ruby-plugin-*.jar"),
         EmbeddedPlugin(Language.XML, "XML", "sonar-xml-plugin-*.jar"),
         EmbeddedPlugin(Language.SECRETS, "Secrets detection", "sonar-text-plugin-*.jar"),
-        EmbeddedPlugin(Language.GO, "Go", "sonar-go-plugin-*.jar")
+        EmbeddedPlugin(Language.GO, "Go", "sonar-go-plugin-*.jar"),
+        EmbeddedPlugin(Language.CLOUDFORMATION, "CloudFormation", "sonar-iac-plugin-*.jar"),
+        EmbeddedPlugin(Language.DOCKER, "Docker", "sonar-iac-plugin-*.jar"),
+        EmbeddedPlugin(Language.KUBERNETES, "Kubernetes", "sonar-iac-plugin-*.jar"),
+        EmbeddedPlugin(Language.TERRAFORM, "Terraform", "sonar-iac-plugin-*.jar")
```
Comment (damien-urruty-sonarsource) [inline src/main/java/org/sonarlint/intellij/core/EmbeddedPlugins.kt:69]: I think we should find a way to simplify that, basically we are defining 4 times the same plugin as embedded

File: src/main/resources/META-INF/plugin.xml [hunk 3]
```diff
@@ -192,17 +192,8 @@
...
-        <externalAnnotator language="PHP" implementationClass="org.sonarlint.intellij.editor.SonarExternalAnnotator"/>
-        <!-- JavaScript is the base language for TypeScript, EcmaScript, jsx, ...-->
-        <externalAnnotator language="JavaScript" implementationClass="org.sonarlint.intellij.editor.SonarExternalAnnotator"/>
-        <externalAnnotator language="ruby" implementationClass="org.sonarlint.intellij.editor.SonarExternalAnnotator"/>
-        <!-- XML is the base language for HTML, XHTML, ...-->
-        <externalAnnotator language="XML" implementationClass="org.sonarlint.intellij.editor.SonarExternalAnnotator"/>
-        <externalAnnotator language="Scala" implementationClass="org.sonarlint.intellij.editor.SonarExternalAnnotator"/>
-        <externalAnnotator language="Swift" implementationClass="org.sonarlint.intellij.editor.SonarExternalAnnotator"/>
+        <!-- Empty language attribute corresponds to any language (see com.intellij.lang.Language.ANY) -->
+        <externalAnnotator language="" implementationClass="org.sonarlint.intellij.editor.SonarExternalAnnotator"/>
```
Comment (henryju) [inline src/main/resources/META-INF/plugin.xml:196]: Youhou!

**Review by henryju — APPROVED** [review]
LGTM

**Pull request merged**